### PR TITLE
NIFIREG-361 - Improved handling of the /access/logout endpoint.

### PR DESCRIPTION
**Pull request by thenatog**

### CODE REVIEW

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js [hunk 1]
```diff
@@ -71,9 +71,11 @@ NfRegistry.prototype = {
      * Invalidate old tokens and route to login page
      */
     logout: function () {
+        this.nfRegistryApi.deleteToLogout().subscribe(function () {
```
Comment (thenatog) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js]: Do I need to do this subscribe? Or do I just called deleteToLogout()?

Comment (alopresto) [reply nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js]: @mcgilman @scottyaslan can you please review this and provide an answer? Thank you.

**Comment by thenatog** [comment]
> @thenatog : looks good in my opinion, and I think the test failure is unrelated, could you confirm?
> 
> ```
> Tests run: 9, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 12.151 sec <<< FAILURE! - in org.apache.nifi.registry.web.api.BucketsIT
> testCreateBucketGetBucket(org.apache.nifi.registry.web.api.BucketsIT)  Time elapsed: 0.124 sec  <<< ERROR!
> javax.ws.rs.ClientErrorException: HTTP 409 Conflict
> 	at org.glassfish.jersey.client.JerseyInvocation.createExceptionForFamily(JerseyInvocation.java:1122)
> 	at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:1105)
> 	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:883)
> 	at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$1(JerseyInvocation.java:767)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:229)
> 	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:414)
> 	at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:765)
> 	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:456)
> 	at org.glassfish.jersey.client.JerseyInvocation$Builder.post(JerseyInvocation.java:357)
> 	at org.apache.nifi.registry.web.api.BucketsIT.testCreateBucketGetBucket(BucketsIT.java:127)
> ```

I'm not seeing this error when I run the tests on my machine. Someone should confirm this.

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js (hunk 1 above)
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js]:
```suggestion
        var self = this;
        this.nfRegistryApi.deleteToLogout().subscribe(
        function () {
             // next call
        },
        function () {
             // error callback
        },
        function () {
              // complete callback... clean up and navigate on complete only
              self.nfStorage.removeItem('jwt');
              delete self.nfRegistryService.currentUser.identity;
              delete self.nfRegistryService.currentUser.anonymous;
              self.router.navigateByUrl('login');
        });
```

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/services/nf-registry.api.js [hunk 2]
```diff
@@ -725,6 +725,37 @@ NfRegistryApi.prototype = {
...
+        var options = {
+            headers: headers,
+            withCredentials: true,
+            responseType: 'text'
+        };
+
+        return this.http.delete('../nifi-registry-api/access/logout', options).pipe(
+            map(function (response) {
+                // remove the token from local storage
+                self.nfStorage.removeItem('jwt');
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/services/nf-registry.api.js]:
```suggestion
               
```

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js [hunk 3]
```diff
@@ -71,9 +71,11 @@ NfRegistry.prototype = {
...
+
+        });
         delete this.nfRegistryService.currentUser.identity;
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js]:
```suggestion
        
```

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js [hunk 4]
```diff
@@ -71,9 +71,11 @@ NfRegistry.prototype = {
...
         delete this.nfRegistryService.currentUser.anonymous;
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js]:
```suggestion
        
```

File: nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js [hunk 5]
```diff
@@ -71,9 +71,11 @@ NfRegistry.prototype = {
...
-        this.nfStorage.removeItem('jwt');
         this.router.navigateByUrl('login');
```
Comment (scottyaslan) [inline nifi-registry-core/nifi-registry-web-ui/src/main/webapp/nf-registry.js]:
```suggestion
        
```

**Comment by alopresto** [comment]
Ran `contrib-check` and all tests pass. +1, merging. Thanks Nathan and Scott.

**Pull request closed without merging**